### PR TITLE
automatically modprobe fiodriver in fio_register

### PR DESCRIPTION
**Pull request by dcrawford1**
I added this change so that we can remove all fio init from the snap `-daemon` start scripts.  Now the fiodriver.ko kernel module is automatically loaded the first time a client calls fio_register

### CODE REVIEW

File: fio/src/fioapi/fio.c [hunk 1]
```diff
@@ -81,6 +81,7 @@ fio_register
 (
 )
 {
+	system("modprobe fiodriver");
```
Comment (dtarico) [inline fio/src/fioapi/fio.c]: This can get called multiple times if multiple apps register with the api. Is that ok?

Comment (gallagjm) [reply fio/src/fioapi/fio.c]: I saw a suggestion to use "/sbin/modprobe" in a thread about starting modules from user space programs, as a more secure option.

**Comment by dcrawford1** [comment]
ok, I will change it to /sbin/modprobe

**Comment by dcrawford1** [comment]
yes, modprobe can be called multiple times and it won't do anything if the module is already loaded

**Review by dtarico — APPROVED** [review]

**Comment by dcrawford1** [comment]
Already manually merged

**Pull request closed without merging**